source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -122,18 +127,28 @@ bool HeaderMatcher::matches(const Network::Connection&,
   return Envoy::Http::HeaderUtility::matchHeaders(headers, header_);
 }
 
-bool IPMatcher::matches(const Network::Connection& connection, const Envoy::Http::RequestHeaderMap&,
-                        const StreamInfo::StreamInfo&) const {
-  const Envoy::Network::Address::InstanceConstSharedPtr& ip =
-      destination_ ? connection.localAddress() : connection.remoteAddress();
-
+bool IPMatcher::matches(const Network::Connection&, const Envoy::Http::RequestHeaderMap&,
+                        const StreamInfo::StreamInfo& info) const {
+  Envoy::Network::Address::InstanceConstSharedPtr ip;
+  switch (type_) {
+  case DownstreamLocal:
+    ip = info.downstreamLocalAddress();
+    break;
+  case DownstreamDirectRemote:
+    ip = info.downstreamDirectRemoteAddress();
```

Comment:
I *think* this is a minor behavior change: in the old code, I believe the `connection.remoteAddress()`, used for proto config `source_ip`, would be informed by proxy-protocol, but it would not be informed by XFF. 

The new behavior makes sense and is consistent with how these two things (proxy proto and XFF) and treated elsewhere, but I'm concerned about the behavior change. 

If we decide this is a behavior change, we may have to leave the existing source_ip behavior as-is, deprecate the field, and add 2 new fields (instead of 1) for this new behavior. 

Sorry I didn't catch this in an earlier review. Can I get another set of eyes from someone who knows the listener filter/proto-protocol code? Maybe @lizan?

Reply:
Yeah I think this is a behavior change. +1 to deprecated source_ip with addition of new behavior.

Reply:
Thanks for the review, I deprecated the `source_ip` with two new fields `peer_ip` and `remote_ip`, let me know if you think it should have a different name.